containers/src/Kokkos_DualView.hpp

```diff
@@ -790,9 +789,7 @@ class DualView : public ViewTraits<DataType, Properties...> {
     if constexpr (impl_dualview_is_single_device) {
       return;
     } else {
-      if (modified_flags.data() == nullptr) {
-        modified_flags = t_modified_flags("DualView::modified_flags");
-      }
+      if (modified_flags.data() == nullptr) return;
```

Comment:
Is there any coverage for this?
I replaced the early return by an abort on my local build and the tests are still passing

Reply:
We probably need a test like this:

```c++
DualView<int*> dv();
dv.modify<host_exec>();
dv.sync<device_exec>();
dv.modify<device_exec>();
dv.sync<host_exec>();
dv.modify_host();
dv.sync_device();
dv.modify_device();
dv.sync_host();
```